docs/proposals/source-verificiation-policies.md

```diff
@@ -0,0 +1,313 @@
+    trustedSigners:
+    - keyID: D56C4FCA57A46444
+    - keyID: 4AEE18F83AFDEB23
+  - repositoryPattern: 'https://github.com/*'
+    verificationLevel: progressive
+    verificationMethod: gpg
+    trustedSigners:
+    - keyID: 4AEE18F83AFDEB23
+```
+
+If the policies would have been defined in the opposite order, the specific policy for `https://github.com/example/super-secure` would never match, because that repository URL is already matched by the other policy's `https://github.com/*` pattern, thus that policy would be applied.
+
+### Creating a new Application from a "mixed-signed" source repository
+
+Having all commits in a Git repository cryptographically signed from the repository's inception is an ideal situation, however the reality often looks different. Often times, the requirement to cryptographically sign commits comes at some point in time of a repository's lifecyle. 
+
+This is what we call "mixed-signed", and is explained in the below diagram:
+
+```
+                   HEAD
+       1.0         2.0
+        |           |
+A---B---C---D---E---F
+-   -   -   +   +   +
+```
+
+Again, the letters `A` through `F` represents commits in the revision history. The dash and plus signs below the commits denote whether a commit is signed (`+`) or unsigned (`-`). So as we can see, the repository is in a "mixed-signed" state, where some commits are signed and others are not.
+
+A revision history such as shown above has a few implications when it comes to commit verification:
+
+1. If an Application under a verification policy that enforces the `strict` verification level tries to sync, it will not be able to succeed, as the `strict` level requires each and every commit in the history to be signed, but commits `A`, `B` and `C` are not.
+
+2. If an Application under a verification policy that enforces the `progressive` verification level tries to sync, it _may_ succeed under certain conditions:
+
+   For example, if the Application is currently synced to revision `C` (potentially before the source verification policy was in place) can sync successfully to target revision `F`, as all of the subsequent commits `D`, `E` and `F` are signed. However, if the Application would currently be synced to revision `B`, it can't sync successfully because `C` is not signed. There just is no way forward.
+   
+   Similarly, if the Application has never synced before (i.e. is a new Application), it won't be able to sync at all, because the verification level `progressive` would behave the same as level `strict`, thus _all_ commits in the history needs to be signed, but commits `A`, `B` and `C` are not.
+   
+3. An Application under a policy which enforces the verification level `head` can sync to any of the commits `D`, `E` and `F` regardless of whether it has synced before.
+
+So as it seems from that explanation of the impact of verification level, a newly Application could only ever be synced with a verification level of `head`. 
+
+### Detailed examples
```

Comment:
If any policy fails validation, the application will not  be synced. One advantage of the new verification policies is that you can define a unique policy on a per-source basis. If one of the sources of a multi-source application is not covered by a policy, it will be automatically a "pass".